app/src/main/java/com/keylesspalace/tusky/db/AccountManager.kt

```diff
@@ -0,0 +1,190 @@
+/* Copyright 2018 Conny Duck
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.db
+
+import android.util.Log
+import com.keylesspalace.tusky.TuskyApplication
+import com.keylesspalace.tusky.entity.Account
+
+/**
+ * This class caches the account database and handles all account related operations
+ * @author ConnyDuck
+ */
+
+class AccountManager {
+
+    var activeAccount: AccountEntity? = null
+
+    private var accounts: MutableList<AccountEntity> = mutableListOf()
+    private val accountDao: AccountDao = TuskyApplication.getDB().accountDao()
+
+    init {
+        accounts = accountDao.loadAll().toMutableList()
+
+        activeAccount = accounts.find { acc ->
+            acc.isActive
+        }
+    }
+
+    /**
+     * Adds a new empty account and makes it the active account.
+     * More account information has to be added later with [updateActiveAccount]
+     * or the account wont be saved to the database.
+     * @param accessToken the access token for the new account
+     * @param domain the domain of the accounts Mastodon instance
+     */
+    fun addAccount(accessToken: String, domain: String) {
+
+        activeAccount?.let{
+            it.isActive = false
+            Log.d("AccountManager", "saving account with id "+it.id)
+
+            accountDao.insertOrReplace(it)
+        }
+
+        activeAccount = AccountEntity(0, domain, "", "", accessToken, "", "", true)
```

Comment:
Not important: I would use named arguments here because it is totally unclear what arguments represent (without IDE at least)